test/runner/runner_test.dart

```diff
@@ -292,6 +292,82 @@ $_usage""");
     // 15078 is fixed.
   });
 
+  group("retries tests", () {
+    test("and eventually passes for valid tests", () {
+      d
+          .file(
+              "test.dart",
+              """
+          import 'dart:async';
+
+          import 'package:test/test.dart';
+
+          int attempt = 0;
+          void main() {
+            test("failure", () {
+             attempt++;
```

Comment:
Long ago there was a proposal for tagged template strings in Dart that could maybe work that way, but it never went anywhere :disappointed:. We could conceivably add a special case to the formatter to look for `"""// Dart` or something, but I doubt @munificent would go for that.